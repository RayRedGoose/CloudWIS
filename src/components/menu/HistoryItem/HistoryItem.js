import React from 'react'
import './HistoryItem.scss'

const HistoryItem = (props) => {
  const changeCityByClick = (event) => {
    const searchCity = event.target.getAttribute('name')
    props.changeCity(searchCity)
    props.toggleActiveState()
    props.fetchWeatherData(searchCity)
  }

  const keyPressed = (event) => {
    if (event.key === "Enter") {
      changeCityByClick(event)
    }
  }

  return (
    <li
      tabIndex="0"
      role="button"
      className="history-item"
      name={props.cityName}
      onClick={changeCityByClick}
      onKeyPress={keyPressed}>{props.cityName}</li>
  )
}

export default HistoryItem
