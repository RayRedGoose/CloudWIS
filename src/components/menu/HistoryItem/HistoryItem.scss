.history-item {
  cursor: pointer;
  font-size: 18px;
  font-weight: 700;
  list-style-type: none;
  background-color: #fff;
  border: 2px solid #0D324D;
  border-radius: 20px;
  color: #0D324D;
  width: fit-content;
  margin-right: 10px;
  margin-bottom: 10px;
  padding: 0.5em 2em;
  &:hover {
    transform: scale(1.1);
  }
  &:active {
    transform: scale(0.9);
  }
}
