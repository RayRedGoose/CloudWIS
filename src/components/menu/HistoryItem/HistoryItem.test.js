import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe("HistoryItem", () => {
  let item, mockProps
  const changeCity = jest.fn()
  const fetchWeatherData = jest.fn()
  const toggleActiveState = jest.fn()

  beforeEach(() => {
    mockProps = {
      changeCity: changeCity,
      fetchWeatherData: fetchWeatherData,
      toggleActiveState: toggleActiveState,
      cityName: 'Denver',
      key: 'Denver'
    }

    item = shallow(
      <HistoryItem {...mockProps}/>
    )
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(item).toMatchSnapshot();
  })

  describe("changeCityByClick", () => {
    let mockEvent

    beforeEach(() => {
      mockEvent = {
        target: {
          getAttribute: jest.fn()
        }
      }
    });

    it('should call changeCity prop when item was pushed', () => {
      item.simulate('click', mockEvent);

      expect(changeCity).toHaveBeenCalled();
    })

    it('should call toggleActiveState prop when item was pushed', () => {
      item.simulate('click', mockEvent);

      expect(toggleActiveState).toHaveBeenCalled();
    })

    it('should call fetchWeatherData prop when item was pushed', () => {
      item.simulate('click', mockEvent);

      expect(fetchWeatherData).toHaveBeenCalled();
    })
  });
});
