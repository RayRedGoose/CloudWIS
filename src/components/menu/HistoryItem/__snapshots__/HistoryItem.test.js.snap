// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem should match the snapshot with all data passed in correctly 1`] = `
<li
  className="history-item"
  name="Denver"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  Denver
</li>
`;
